### DISCUSSION TITLE
Menu with Select and Checkbox

### DISCUSSION THREAD

**Discussion by ldalves** [mantinedev/mantine#1760] [Q&A] [answered]
Hi! 👋 

Is there any best practice on how to have other components like `Select` and `Checkboxes` inside a `Menu` component?

When I wrap them inside a `Menu.Item` and try to interact with the internal components, the main `Menu` closes on click.

I would like to know if there is any recommendation on how to do this correctly.

Best regards, 

Leandro

**Comment by rtivital** [comment, marked as answer]
Menu component does not support interactive elements as menu items. If you need these features, you should use Popover component instead.
